compiler/src/iree/compiler/Codegen/Utils/Utils.h

```diff
@@ -43,28 +43,40 @@ FailureOr<IREE::HAL::ExecutableExportOp> getEntryPoint(func::FuncOp funcOp);
 /// failure.
 FailureOr<IREE::HAL::ExecutableVariantOp> getExecutableVariantOp(Operation *op);
 
-/// Methods to get target information.
-bool isX86(IREE::HAL::ExecutableVariantOp variantOp);
+/// Returns the StringAttr with the name `stringAttr` in the `targetAttr`, if
+/// found.
+Optional<StringAttr> getConfigStringAttr(
+    IREE::HAL::ExecutableTargetAttr targetAttr, StringRef stringAttr);
 
-bool isAArch64(IREE::HAL::ExecutableVariantOp variantOp);
+/// Returns the IntegerAttr with the name `integerAttr` in the `targetAttr`, if
+/// found.
+Optional<IntegerAttr> getConfigIntegerAttr(
+    IREE::HAL::ExecutableTargetAttr targetAttr, StringRef integerAttr);
 
-bool isRISCV(IREE::HAL::ExecutableVariantOp variantOp);
+/// Returns the LLVM Target triple associated with the `targetAttr`, if set.
+Optional<llvm::Triple> getTargetTriple(
+    IREE::HAL::ExecutableTargetAttr targetAttr);
 
-inline bool isVMVXBackend(IREE::HAL::ExecutableVariantOp variantOp) {
-  return variantOp.getTarget().getBackend().getValue().startswith("vmvx");
-}
+/// Returns the CPU target features associated with the `targetAttr`, if set.
+Optional<StringRef> getCpuFeatures(IREE::HAL::ExecutableTargetAttr targetAttr);
+
+/// Methods to get target information.
+bool isX86(IREE::HAL::ExecutableTargetAttr targetAttr);
+bool isAArch64(IREE::HAL::ExecutableTargetAttr targetAttr);
+bool isRISCV(IREE::HAL::ExecutableTargetAttr targetAttr);
+bool isVMVXBackend(IREE::HAL::ExecutableTargetAttr targetAttr);
 
 /// Returns true if the 'variantOp' contains '+avx2' in its cpu features.
-bool hasAVX2Feature(IREE::HAL::ExecutableVariantOp variantOp);
+bool hasAVX2Feature(IREE::HAL::ExecutableTargetAttr targetAttr);
```

Comment:
(I know you're just touching these lines and didn't make them)
It's a bit smelly that this shared utils file has so many backend-specific things in it - "isX86" and "hasAVX2Feature" are questions that only LLVMCPU/ should be asking and don't really belong in a shared utils directory. It'd be good for someone to re-evaluate this dumping ground at some point before this becomes our thousand-line pile of random backend-specific things all mixed together.

Reply:
+1 to this. @dcaballe and @hanhanW to maybe move these into `LLVMCPU/` folder... no blame, I probably reviewed and approved it....